lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/transport/netty/NettyMessagingTransport.java

```diff
@@ -111,6 +117,7 @@
    * @param numberOfTries the number of tries of connection
    * @param retryTimeout  the timeout of reconnection
    * @param tcpPortProvider  gives an iterator that produces random tcp ports in a range
+   * @param protocolType  the protocol to use for transport
    */
   @Inject
   private NettyMessagingTransport(
```

Comment:
This class contains lots of if else. I'm not sure this is a good design. We can remove the `if`...`else` by receiving variables from the constructor.